src/session/content/message_row/reply.rs

```diff
@@ -0,0 +1,222 @@
+                        .read_only()
+                        .explicit_notify()
+                        .build(),
+                ]
+            });
+            PROPERTIES.as_ref()
+        }
+
+        fn property(&self, _id: usize, pspec: &glib::ParamSpec) -> glib::Value {
+            let obj = self.obj();
+
+            match pspec.name() {
+                "chat" => obj.imp().chat.borrow().clone().to_value(),
+                "reply-id" => obj.imp().reply_id.get().to_value(),
+                _ => unimplemented!(),
+            }
+        }
+
+        fn set_property(&self, _id: usize, value: &glib::Value, pspec: &glib::ParamSpec) {
+            let obj = self.obj();
+
+            match pspec.name() {
+                "chat" => obj.set_chat(value.get().unwrap()),
+                "reply-id" => obj.imp().reply_id.set(value.get().unwrap()),
+                _ => unimplemented!(),
+            }
+        }
+
+        fn constructed(&self) {
+            self.is_loading.set(true);
+            self.message_label
+                .set_label(&gettextrs::gettext("Loading ..."));
+        }
+
+        fn dispose(&self) {
+            while let Some(child) = self.obj().first_child() {
+                child.unparent();
+            }
+        }
+    }
+
+    impl WidgetImpl for MessageReply {}
+    impl BinImpl for MessageReply {}
+    impl MessageBaseImpl for MessageReply {}
+}
+
+glib::wrapper! {
+    pub(crate) struct MessageReply(ObjectSubclass<imp::MessageReply>)
+        @extends gtk::Widget, adw::Bin;
+}
+
+impl MessageReply {
+    pub(crate) fn set_chat(&self, chat: Option<Chat>) {
+        if self.imp().chat.borrow().clone() == chat {
+            return;
+        }
+
+        self.imp().chat.replace(chat);
+        self.notify("chat");
+    }
+
+    pub(crate) fn set_reply_id(&self, reply_id: i64) {
+        let imp = self.imp();
+
+        if imp.reply_id.get() == reply_id {
+            return;
+        }
+
+        imp.reply_id.set(reply_id);
+        self.notify("reply-id");
+    }
+
+    pub(crate) fn load_message(&self) {
+        let imp = self.imp();
+
+        let reply_id = imp.reply_id.get();
+        let chat = imp.chat.borrow().clone().unwrap();
+
+        if let Some(message) = chat.message(reply_id) {
+            self.update_from_message(&message);
+        } else {
+            spawn(clone!(@weak self as obj => async move {
+                let chat = obj.imp().chat.borrow().clone().unwrap();
+                if let Some(message) = chat.fetch_chat_message(reply_id).await {
+                    obj.update_from_message(&message);
+                } else {
+                    // Message doesn't exist, so we should remove "Loading..." caption
+                    // TODO: Impelent it properly using signals
+                    obj.imp().message_label.set_label("Deleted message");
+                }
+            }));
+        }
+
+        imp.is_loading.set(false);
+        self.notify("is-loading");
```

Comment:
This is causing the following warning:

`(telegrand:2): GLib-GObject-WARNING **: 22:57:20.401: g_object_notify: object class 'MessageReply' has no property named 'is-loading'`